src/ixdat/techniques/cv.py

```diff
@@ -106,3 +120,216 @@ def redefine_cycle(self, start_potential=None, redox=None, N_points=5):
+            )
+            if spec[1] in ["anodic", "cathodic"]
+        ]
+        others_sweep_specs = [
+            spec
+            for spec in other.get_timed_sweeps(
+                v_scan_res=v_scan_res, res_points=res_points
+            )
+            if spec[1] in ["anodic", "cathodic"]
+        ]
+        if not len(my_sweep_specs) == len(others_sweep_specs):
+            raise BuildError(
+                "Can only make diff of CyclicVoltammograms with same number of sweeps."
+                f"{self} has {my_sweep_specs} and {other} has {others_sweep_specs}."
+            )
+
+        diff_values = {name: np.array([]) for name in v_list}
+        t_diff = np.array([])
+
+        for my_spec, other_spec in zip(my_sweep_specs, others_sweep_specs):
+            sweep_type = my_spec[1]
+            if not other_spec[1] == sweep_type:
+                raise BuildError(
+                    "Corresponding sweeps must be of same type when making diff."
+                    f"Can't align {self}'s {my_spec} with {other}'s {other_spec}."
+                )
+            my_tspan = my_spec[0]
+            other_tspan = other_spec[0]
+            my_t, my_potential = self.grab(
+                "potential", my_tspan, include_endpoints=False
+            )
+            t_diff = np.append(t_diff, my_t)
+            other_t, other_potential = other.grab(
+                "potential", other_tspan, include_endpoints=False
+            )
+            if sweep_type == "anodic":
+                other_t_interp = np.interp(
+                    np.sort(my_potential), np.sort(other_potential), other_t
+                )
+            elif sweep_type == "cathodic":
+                other_t_interp = np.interp(
+                    np.sort(-my_potential), np.sort(-other_potential), other_t
+                )
+            else:
+                continue
+            for name in v_list:
+                my_v = self.grab_for_t(name, my_t)
+                other_v = other.grab_for_t(name, other_t_interp)
+                diff_v = my_v - other_v
+                diff_values[name] = np.append(diff_values[name], diff_v)
+
+        t_diff_series = TimeSeries(
+            name="time/[s] for diffs", unit_name="s", data=t_diff, tstamp=self.tstamp
+        )  # I think this is the same as self.potential.tseries
+
+        series_list.append(t_diff_series)
+        for name, data in diff_values.items():
+            series_list.append(
+                ValueSeries(
+                    name=name,
+                    unit_name=self[name].unit_name,
+                    data=data,
+                    tseries=t_diff_series,
+                )
+            )
+
+        diff_as_dict = self.as_dict()
+        del diff_as_dict["s_ids"]
+
+        diff_as_dict["series_list"] = series_list
+
+        cls = cls or CyclicVoltammogramDiff
+        diff = cls.from_dict(diff_as_dict)
+        diff.cv_1 = self
+        diff.cv_2 = other
+        return diff
```

Comment:
With regards to https://github.com/ixdat/ixdat/pull/30#discussion_r811203971

I wonder if this is the source of the confusion of what `cv_1` and `cv_2`? As you see, they are set outside the `CyclicVoltammogramDiff`'s methods. That may not be ideal. I think the right way to do this would be to pass them on through dictionary representations and evt. the `MemoryBackend`, but would rather wait on that until after the "metaprogramming" work. I'll put a TODO here linking to the chat.

Reply:
Ahhh, then I understand. Well, of you might just call them `cv_compare_1` and `cv_compare_2`, which would probably have started my thinking along those lines. In any case, agree that it can be deferred.

Reply:
renamed to `cv_compare_1` and `cv_compare_2`.